adafruit_rgb_display/rgb.py

```diff
@@ -71,74 +82,90 @@ def _decode_pixel(self, data):
-            return self._decode_pixel(self._block(x, y, x, y))
-        if not 0 <= x < self.width or not 0 <= y < self.height:
-            return
-        self._block(x, y, x, y, self._encode_pixel(color))
+            return self._decode_pixel(self._block(xpos, ypos, xpos, ypos))
 
-    def fill_rectangle(self, x, y, width, height, color):
+        if 0 <= xpos < self.width and 0 <= ypos < self.height:
+            self._block(xpos, ypos, xpos, ypos, self._encode_pixel(color))
+        return None
+
+    #pylint: disable-msg=too-many-arguments
+    def fill_rectangle(self, xpos, ypos, width, height, color):
         """Draw a filled rectangle."""
-        x = min(self.width - 1, max(0, x))
-        y = min(self.height - 1, max(0, y))
-        w = min(self.width - x, max(1, width))
-        h = min(self.height - y, max(1, height))
-        self._block(x, y, x + w - 1, y + h - 1, b'')
-        chunks, rest = divmod(w * h, _BUFFER_SIZE)
+        xpos = min(self.width - 1, max(0, xpos))
+        ypos = min(self.height - 1, max(0, ypos))
+        width = min(self.width - xpos, max(1, width))
+        height = min(self.height - ypos, max(1, height))
+        self._block(xpos, ypos, xpos + width - 1, ypos + height - 1, b'')
+        chunks, rest = divmod(width * height, _BUFFER_SIZE)
         pixel = self._encode_pixel(color)
         if chunks:
             data = pixel * _BUFFER_SIZE
-            for count in range(chunks):
-                self._write(None, data)
-        self._write(None, pixel * rest)
+            for _ in range(chunks):
+                self.write(None, data)
+        self.write(None, pixel * rest)
+    #pylint: enable-msg=too-many-arguments
 
     def fill(self, color=0):
         """Fill whole screen."""
         self.fill_rectangle(0, 0, self.width, self.height, color)
 
-    def hline(self, x, y, width, color):
+    def hline(self, xpos, ypos, width, color):
         """Draw a horizontal line."""
-        self.fill_rectangle(x, y, width, 1, color)
+        self.fill_rectangle(xpos, ypos, width, 1, color)
 
-    def vline(self, x, y, height, color):
+    def vline(self, xpos, ypos, height, color):
         """Draw a vertical line."""
-        self.fill_rectangle(x, y, 1, height, color)
+        self.fill_rectangle(xpos, ypos, 1, height, color)
+
+    def write(self, command, data):
+        """Derived class must implement this"""
+        raise NotImplementedError
 
+    def read(self, command, count):
+        """Derived class must implement this"""
+        raise NotImplementedError
 
 class DisplaySPI(Display):
+    """Base class for SPI type devices"""
+    #pylint: disable-msg=too-many-arguments
     def __init__(self, spi, dc, cs, rst=None, width=1, height=1, baudrate=1000000,
                  polarity=0, phase=0):
         self.spi_device = spi_device.SPIDevice(spi, cs, baudrate=baudrate,
                                                polarity=polarity, phase=phase)
-        self.dc = dc
+        self.d_or_c = dc
```

Comment:
I believe with did this one in the SPI already. "d_or_c" makes no sense. Please use "dc_pin" instead.

Reply:
changed to dc_pin